src/react/utils.js

```diff
@@ -235,6 +239,33 @@ export function forEachArrayValue(realm: Realm, array: ObjectValue, mapFunc: Fun
   }
 }
 
+export function mapArrayValue(
+  realm: Realm,
+  array: ArrayValue,
+  mapFunc: (element: Value, descriptor: Descriptor) => Value
+): ArrayValue {
+  let lengthValue = Get(realm, array, "length");
+  invariant(lengthValue instanceof NumberValue, "Invalid length on ArrayValue during reconcilation");
+  let length = lengthValue.value;
+  let newArray = Create.ArrayCreate(realm, length);
+  let returnTheNewArray = false;
+
+  for (let i = 0; i < length; i++) {
+    let elementProperty = array.properties.get("" + i);
+    let elementPropertyDescriptor = elementProperty && elementProperty.descriptor;
+    invariant(elementPropertyDescriptor, `Invalid ArrayValue[${i}] descriptor`);
+    let elementValue = elementPropertyDescriptor.value;
+    if (elementValue instanceof Value) {
```

Comment:
Is it okay to just silently drop elements with getters/setters?

Reply:
Yes, these are ReactElement children – it's created via an internal part of React.